task.py

```diff
@@ -381,7 +382,21 @@ def get_pod_ip(self) -> str:
         pod_ip = None
         try:
             if y:
-                pod_ip = y["status"]["podIP"]
+                if self.ts.connection.secondary_network_nad:
```

Comment:
This will currently always be True, because `secondary_network_nad` is filled with the default `f"{namespace}/ocp-secondary"`.

If you need to differentiate whether the caller explicitly configured the setting in YAML, you must make `secondary_network_nad: Optional[str]`, so that a `None` represents unset (and each callers need to figure out what actual value to use).

Reply:
Ack I will do that.